### PR TITLE
LPS-196141 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-182060 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-196141](https://liferay.atlassian.net/browse/LPS-196141) → [LPS-196141](https://liferay.atlassian.net/browse/LPS-196141)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 1]
```diff
@@ -319,4 +319,79 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182060. Time without autosave is a number between 1 and 99999."
+	@priority = 3
+	test ViewTimeWithoutAutosave {
+		property custom.properties = "lock.expiration.time.com.lliferay.portal.kernel.model.Layout=6000";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:325]: Deprecated property

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminAccessibility.testcase [hunk 2]
```diff
@@ -121,4 +121,39 @@ definition {
...
+				value1 = 0);
+
+			Button.clickSave();
+		}
+
+		task ("Then an error message is displayed") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "TextInput#REQUIRED_ALERT",
+				value1 = "Please enter a value greater than or equal to 1.");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminAccessibility.testcase:152]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminAccessibility.testcase [hunk 3]
```diff
@@ -121,4 +121,39 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182060. The Locked Pages admin should comply with WCAG 2.1 AA standard."
+	@priority = 4
+	test ViewLockedPagesInstanceSettingsAccessibility {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminAccessibility.testcase:126]: Rename to ViewAccessibilityOfLockedPagesUnderInstanceSettings

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 4]
```diff
@@ -21,6 +21,105 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182060. Allow Automatic Unlocking Process can be enabled and disabled  via keyboard."
+	@priority = 4
+	test EnableAllowAutomaticUnlockingProcessViaKeyboard {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:26]: Rename to EnableAllowAutomaticUnlockingProcessUnderInstanceSettingsViaKeyboard

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 5]
```diff
@@ -319,4 +319,79 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-182060. Time without autosave is a number between 1 and 99999."
+	@priority = 3
+	test ViewTimeWithoutAutosave {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:324]: Rename to ConfigureTimeWithoutAutosaveUnderInstanceSettings

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 6]
```diff
@@ -21,6 +21,105 @@ definition {
...
+			SystemSettings.gotoConfiguration(
+				configurationCategory = "Pages",
+				configurationName = "Locked Pages",
+				configurationScope = "Virtual Instance Scope");
+		}
+
+		task ("When focusing on the Allow Automatic Unlocking Process toggle via keyboard") {
+			KeyPress(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:42]: Use previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 7]
```diff
@@ -21,6 +21,105 @@ definition {
...
+
+			AssertNotChecked.assertNotCheckedNotVisible(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
+
+		task ("When focusing on the Allow Automatic Unlocking Process toggle via keyboard") {
+			KeyPress(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:60]: Use previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 8]
```diff
@@ -21,6 +21,105 @@ definition {
...
+			AssertChecked.assertCheckedNotVisible(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
+
+		task ("When filling in the Time Without Autosave field") {
+			KeyPress(
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = "\TAB");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:79]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 9]
```diff
@@ -21,6 +21,105 @@ definition {
...
+			Type(
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = 7);
+		}
+
+		task ("Then it is possible to save changes via keyboard") {
+			KeyPress(
+				key_text = "Save",
+				locator1 = "Button#ANY",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:90]: Use previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 10]
```diff
@@ -21,6 +21,105 @@ definition {
...
+
+			KeyPress(
+				key_text = "Save",
+				locator1 = "Button#ANY",
+				value1 = "\ENTER");
+
+			AssertTextEquals(
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = 7);
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:101]: Alert.viewSuccessMessage

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 11]
```diff
@@ -21,6 +21,105 @@ definition {
...
+		}
+
+		task ("When focusing on the Cancel button via keyboard") {
+			KeyPress(
+				key_text = "Cancel",
+				locator1 = "Button#ANY",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:107]: Use previous element

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 12]
```diff
@@ -21,6 +21,105 @@ definition {
...
+
+		task ("Then it is possible to return to the Instance Settings page via keyboard") {
+			KeyPress(
+				key_text = "Cancel",
+				locator1 = "Button#ANY",
+				value1 = "\ENTER");
+
+			AssertElementPresent(
+				key_title = "Instance Settings",
+				locator1 = "Header#H1_TITLE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:119]: ControlMenu.viewHeaderTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 13]
```diff
@@ -319,4 +319,79 @@ definition {
...
+
+		task ("When trying to leave the field blank") {
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Type(
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = "");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:347]: No need

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, could you update according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 14]
```diff
@@ -319,4 +319,79 @@ definition {
...
+
+			Button.clickSave();
+
+			Alert.viewSpecificRequiredField(fieldFieldLabel = "Time Without Autosave");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:351]:
```
				AssertVisible(
					key_alertMessage = "The Time Without Autosave field is required.",
					locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 15]
```diff
@@ -319,4 +319,79 @@ definition {
...
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = "WebContentTitle@!&#-en_US-es_ES.xlf");
+		}
+
+		task ("Then an error message is displayed") {
+			AssertTextEquals(
+				key_fieldLabel = "Time Without Autosave",
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = "");
```
[the remark this answers is not in the document]
Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:363]:
```
				AssertVisible(
					key_alertMessage = "The Time Without Autosave field is required.",
					locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 16]
```diff
@@ -319,4 +319,79 @@ definition {
...
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = 0);
+
+			Button.clickSave();
+		}
+
+		task ("Then an error message is displayed") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "TextInput#REQUIRED_ALERT",
+				value1 = "Please enter a value greater than or equal to 1.");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:378]:
```
				AssertVisible(
					key_alertMessage = "Please enter a value greater than or equal to 1.",
					locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase [hunk 17]
```diff
@@ -319,4 +319,79 @@ definition {
...
+				locator1 = "TextInput#GENERIC_NUMBER_INPUT",
+				value1 = 200000);
+
+			Button.clickSave();
+		}
+
+		task ("Then an error message is displayed") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "TextInput#REQUIRED_ALERT",
+				value1 = "Please enter a value less than or equal to 99999.");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/lockedpages/LockedPages.testcase:393]:
```
				AssertVisible(
					key_alertMessage = "Please enter a value less than or equal to 99999.",
					locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
```

**Review by Tim-Cao — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase [hunk 18]
```diff
@@ -22,6 +22,91 @@ definition {
...
+				value1 = "\SPACE");
+
+			AssertNotChecked.assertNotCheckedNotVisible(
+				key_toggleSwitchLabel = "Allow Automatic Unlocking Process",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+		}
+
+		task ("When focusing on the Allow Automatic Unlocking Process toggle via keyboard") {
+			KeyPress(
+				locator1 = "//label[contains(@class,'toggle-switch')]",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:59]: The previous element should be 
image

@rafaellajordao

**Comment by Tim-Cao** [comment]
@rafaellajordao, see changes https://github.com/liferay-echo/liferay-portal/pull/14440

**Comment by Tim-Cao** [comment]
ci:close

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase (hunk 18 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/LockedPagesAdminWithKeyboard.testcase:59]: Ohhh now I understand!
I was understanding that if I was using a //span I should change to using a //div, I had understood that this was the "previous element".
Thank you very much Tim, I promise that next time I will do it the right way! @Tim-Cao

**Pull request closed without merging**